### PR TITLE
CI and Build Scripts

### PR DESCRIPTION
**Pull request by ChrisMaddock**
CI and build scripts for the F# Samples.

- Build scripts are as discussed in nunit/nunit-vb-samples#4 and nunit/nunit-csharp-samples#6

Other changes include: 
- Update of NUnit package from alpha
- Change assembly names to match fsproj names - cake script expects this, and I didn't see any reason why they shouldn't match.
- FSharp.Core is now fetched from NuGet, due to a possible packaging bug with mono (See: http://stackoverflow.com/q/31866034/1768779) F# 2.3 isn't available on NuGet, so the package is the latest F# 3.X version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+language: csharp
```
Comment (ChrisMaddock) [inline .travis.yml]: csharp keyword is required for f# projects on Travis - not a mistake!

**Comment by rprouse** [comment]
I have enabled the AppVeyor and Travis builds. If they don't require any special setup, push a small change to trigger builds.

**Comment by ChrisMaddock** [comment]
@rprouse: Done, and passed. :-) (Would you mind enabling the hooks on the C++ samples too, in advance? Planning on doing the same over there, sometime soon.)

**Comment by rprouse** [comment]
Build output looks good on both so I am merging. I have enabled the C++ samples on both CI platforms for when you are ready.

**Comment by ChrisMaddock** [comment]
Great - thanks @rprouse!

**Pull request merged**